app/controllers/courses_controller.rb

```diff
@@ -12,21 +18,43 @@ def show
   end
 
   def update
+    if !current_user.admin?
```

Comment:
Might even be worth needing to explicitly white-list certain routes for students. We can talk about it.